### PR TITLE
Suggested edits to data

### PR DESCRIPTION
**Pull request by efeichen**
Hello there! I used Mavo to suggest the following edits:
Add documentation for contains, get, from, fromlast, to, tofirst, filename, json, and log.
Preview my changes here: https://mavo.io/docs/functions/?functions-storage=https%3A%2F%2Fgithub.com%2Fmavoweb%2Fmavo.io%2Fdocs%2Ffunctions%2Ffunctions.json

### CODE REVIEW

**Comment by efeichen** [comment]
Some questions I have about documenting:

1. Besides returning the text after the last forward slash of an URL, does `filename()` also return the value of a Mavo property? I tried `filename(name)` on the Talks List website and it returns me a conference name.
2. I'm wondering how regular Mavo users would use `json()` and `log()` since they kind of touches a little bit of JS and my descriptions of these two functions have to use some JS terminologies.
3. How should I document the `meta` parameter of `get()`? Does the `meta` object have any other capabilities besides storing the name of the found key or index of the found array item?

**Comment by LeaVerou** [comment]
1. If it's not a URL, it just returns the value as-is.
2. They are for power users. `json()` converts the data passed to it to JSON. `log()` logs values to the console.
3. You shouldn't, that's for internal use.

**Comment by LeaVerou** [comment]
Kenneth, did you see my comments? I see that the meta parameter of get is still documented.

File: docs/functions/functions.json [hunk 1]
```diff
@@ -1346,6 +1470,30 @@
 				true
 			]
 		},
+		{
+			"name": "contains",
+			"id": "contains",
+			"unary": false,
+			"category": "math",
+			"role": "function",
+			"description": "Check if a collection, group, list, or text contains a needle. For example, `contains(group(name:\"Kenneth\", age:17, food:\"rice\"), \"rice\")` returns `true`.\nIf both haystack and needle are a list, `contains()` checks if the first item of the haystack contains the first item of the needle, and so on, and returns a list of corresponding results. For example, `contains(list(\"Kenneth\", \"Verou\"), list(\"Ken\", \"Lea\"))` will return `list(true, false)`.",
```
Comment (LeaVerou) [inline docs/functions/functions.json]: A needle? What is a needle? Is this about making web apps or sewing? :) 
"Check if a value contains some given text" or something like that may make more sense.

Also, you may want a simpler example as the first one, with just text.

File: docs/functions/functions.json [hunk 2]
```diff
@@ -1424,6 +1572,69 @@
...
+			"returnType": "Text",
+			"returnDescription": "Text after the rightmost slash of an URL. Returns the value of the parameter as-is if not an URL."
+		},
+		{
+			"name": "json",
+			"id": "json",
+			"unary": false,
+			"category": "other",
+			"role": "function",
+			"description": "Converts data into JSON string and avoids circular reference.",
```
Comment (LeaVerou) [inline docs/functions/functions.json]: The part about circular references is confusing to novices, this is not something they should worry about, I'd suggest removing it.
Also, what's a "JSON string"? You mean a string of JSON, but also any quoted and properly escaped text is technically a "JSON string". Just say it converts data to JSON, that's perfectly sufficient and not ambiguous!
Same applies for all your other mentions of "JSON string". JSON is a text-based format by itself, you don't need to add "string" after it. That's like documenting any function that produces HTML as producing an "HTML string".

File: docs/functions/functions.json [hunk 3]
```diff
@@ -1424,6 +1572,69 @@
...
+			"returnType": "Text",
+			"returnDescription": "The JSON string."
+		},
+		{
+			"name": "log",
+			"id": "log",
+			"unary": false,
+			"category": "other",
+			"role": "function",
+			"description": "Logs the result of all parameters to the browser console and returns the result of the first parameter of this function.\nFor example, `log(1 + 1, floor(3.5), true && false)` will return `2`, the result of the first parameter of `log()`. Additionally, when you open the browser console, `2 3 false` will be displayed.",
```
Comment (LeaVerou) [inline docs/functions/functions.json]: Maybe mention that this is primarily intended for advanced users and developers and link to the Mavo inspector as something that may be more useful?

File: docs/functions/functions.json [hunk 4]
```diff
@@ -368,6 +368,34 @@
...
+					"description": "Where to look up keys and find values."
+				},
+				{
+					"name": "key",
+					"optional": false,
+					"type": "Text",
+					"description": "The key to look for or the condition to match if in the form `\"key=value\"`."
+				},
+				{
+					"name": "meta",
```
Comment (LeaVerou) [inline docs/functions/functions.json]: As I mentioned in my comment, please remove this. It's for internal use.

**Comment by LeaVerou** [comment]
FWIW the [Google Sheets docs](https://support.google.com/docs/table/25273?hl=en) seem to use "string" and "text", which I think is confusing, since they basically refer to the same thing (both are strings — or text!).

File: docs/functions/functions.json (hunk 4 above)
Comment (efeichen) [reply docs/functions/functions.json]: OK. I did see your comment though. Maybe I forgot to remove it when I was editing.

File: docs/functions/functions.json (hunk 1 above)
Comment (efeichen) [reply docs/functions/functions.json]: Something like: `contains("Kenneth", "Ken")` returns `true`? If so, I feel like it won't demonstrate the core functionality of `contains` very well. Some viewers might think, "Wait, isn't that similar to `search`"?

**Comment by LeaVerou** [comment]
> Something like: `contains("Kenneth", "Ken")` returns true? If so, I feel like it won't demonstrate the core functionality of contains very well. Some viewers might think, "Wait, isn't that similar to search"?

I didn't suggest including **only** that example, just that **as well**.
I'm gonna go ahead and merge this now, feel free to send updates!

**Pull request merged**